src/test/regress/sql/multi_mx_node_metadata.sql

```diff
@@ -27,6 +27,20 @@ CREATE FUNCTION mark_node_readonly(hostname TEXT, port INTEGER, isreadonly BOOLE
                              ARRAY['SELECT pg_reload_conf()'], false);
 $$;
 
+CREATE OR REPLACE FUNCTION trigger_metadata_sync()
+    RETURNS void
+    LANGUAGE C STRICT
+    AS 'citus';
+
+CREATE FUNCTION wait_until_process_count(appname text, target_count int) returns void AS $$
+declare
+   counter integer := -1;
+begin
+ while counter != target_count loop
+  SELECT into counter FROM pg_stat_activity WHERE application_name = appname;
```

Comment:
i'm not sure how this worked, but this should be SELECT count(*) into counter ...

Reply:
I'm guessing this worked because it reached NULL at some point. Agree it should be count(*).